examples/platform/silabs/ICDSubscriptionCallback.cpp

```diff
@@ -0,0 +1,52 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "ICDSubscriptionCallback.h"
+#include <platform/CHIPDeviceConfig.h>
+
+CHIP_ERROR ICDSubscriptionCallback::OnSubscriptionRequested(chip::app::ReadHandler & aReadHandler,
+                                                            chip::Transport::SecureSession & aSecureSession)
+{
+    using namespace chip::System::Clock;
+
+    Seconds32 interval_s32 = std::chrono::duration_cast<Seconds32>(CHIP_DEVICE_CONFIG_SED_IDLE_INTERVAL);
+
+    if (interval_s32 > Seconds16::max())
+    {
+        interval_s32 = Seconds16::max();
+    }
+    uint16_t decidedMaxInterval = interval_s32.count();
+
+    uint16_t requestedMinInterval = 0;
+    uint16_t requestedMaxInterval = 0;
+    aReadHandler.GetReportingIntervals(requestedMinInterval, requestedMaxInterval);
+
+    // If requestedMinInterval is greater than IdleTimeInterval, select next wake up time as max interval
+    if (requestedMinInterval > decidedMaxInterval)
+    {
+        uint8_t ratio = requestedMinInterval / decidedMaxInterval;
+        if (requestedMinInterval % decidedMaxInterval)
+        {
+            ratio++;
+        }
+
+        decidedMaxInterval *= ratio;
```

Comment:
What happens here if `decidedMaxInterval > Seconds16::max()/2` and `requestedMinInterval > decidedMaxInterval`?  Looks to me like we will overflow.

Reply:
`decidedMaxInterval` initial value will never be greater than 3600 since the maximum spec allowed sleep time is an hour.

For the overflow case to happen, the `requestedMinInterval` would need to be greater than :
```
floor(Seconds16::max() / decidedMaxInterval) * decidedMaxInterval
```
The smallest possible `requestedMinInterval` with the longuest allowed ICD sleep time would be :
```
floor(65535 / 3600) * devcidedMaxInterval =  64800s = 18h.
```
So it's technically possible to have an overflow but it seems almost impossible for it to happen.

Reply:
@bzbarsky-apple I added more checks to always have a valid maxInterval